### ISSUE TITLE
Lab2: Failed to create LambdaExecutionRoles

### ISSUE THREAD

**Issue by mervinyan** [aws-samples/aws-serverless-saas-workshop#58]
I followed the instructions listed in the workshop but the cloudfromation was failed in lab2.

Below are the errors I saw in the cloudfromation events tab:

The following resource(s) failed to create: [CreateUserLambdaExecutionRole, AuthorizerExecutionRole, RegisterTenantLambdaExecutionRole, TenantManagementLambdaExecutionRole, TenantUserPoolLambdaExecutionRole].

The security token included in the request is invalid (Service: AmazonIdentityManagement; Status Code: 403; Error Code: InvalidClientTokenId; Request ID: 407dadf1-07d6-4a71-ac6a-f2d5bcd6e8f3; Proxy: null)

Any idea on how to fix it?

Thanks!

**Comment by ujwalbukka** [comment]
@mervinyan - you might have seen this error in the cloud9 instance as Cloudformation template might be taking more time to deploy and the request's token might have expired. 
If you see under Cloudformation service, you will that the cloudfromation template will be still deploying or already successfully deployed. 
If you are unclear of what it has deployed, you can just rerun the same step.

**Comment by mervinyan** [comment]
Hi @ujwalbukka I checked the cloudformation service the deployment failed actually... see the screenshot below
 
![imageedit_3_5180818755](https://github.com/aws-samples/aws-serverless-saas-workshop/assets/1260845/e501b454-7291-460d-806c-7df9ba92e52d)

**Comment by ujwalbukka** [comment]
Hi @mervinyan - In the CloudFormation under Deleted stacks can you check what error "serverless-saas-LambdaFunctions" nested stack has thrown?

You can look at deleted stacks as shown below
![Snip20230706_5](https://github.com/aws-samples/aws-serverless-saas-workshop/assets/63979260/1e26d14d-7d8b-460e-b6f5-6699bc9b68a9)

**Comment by mervinyan** [comment]
Hi @ujwalbukka 
![2023-07-06_19-58](https://github.com/aws-samples/aws-serverless-saas-workshop/assets/1260845/20f44950-3efa-4f01-80b5-3360de3d6216)

**Comment by ujwalbukka** [comment]
Hi @mervinyan 

Have you tried re-running this and still seeing same error consistently?

If yes, 

1. Can you check the client(assuming you are using cloud9) from where you are deploying this stack has proper role permissions and/or unexpired credentials.
2. Do you see any of the nested stacks(maybe **serverless-saas-UserInterface**) taking more time to deploy and that might leading to the expired tokens.

**Comment by mervinyan** [comment]
Hi @ujwalbukka 

Yes I have tried few times and seen the same error consistently. 

How do I check whether the client (cloud9) has proper role permissions and/or unexperied crenditals? Should I check the ./aws/crendtials file?

The serverless-saas-UserInterface stack deployment has taken  roughly 5 minutes in my last run.
![2023-07-07_09-38_1](https://github.com/aws-samples/aws-serverless-saas-workshop/assets/1260845/b64e779d-4685-4f52-b7a5-c35b75087ed3)
![2023-07-07_09-38](https://github.com/aws-samples/aws-serverless-saas-workshop/assets/1260845/8bb15539-577b-4b15-8816-dd55c37da9a4)

Thanks in advance for your help!

**Comment by ujwalbukka** [comment]
Hi @mervinyan 

Can you try doing below things and then retry

1. You need to make sure the role attached to cloud9 EC2 instance has proper permissions. 
           a.  Follow through this [link](https://catalog.us-east-1.prod.workshops.aws/workshops/e04c0885-830a-479b-844b-4c7af79697f8/en-US/lab-0/2-on-your-own/3-create-iam-role) to create a IAM role with admin access
            b. Follow through this [link](https://catalog.us-east-1.prod.workshops.aws/workshops/e04c0885-830a-479b-844b-4c7af79697f8/en-US/lab-0/2-on-your-own/4-attach-iam-role) to attach the role you created to cloud9

2. Disable "AWS managed temporary credentials" in the Cloud9 settings
         a. Follow through this [link](https://catalog.us-east-1.prod.workshops.aws/workshops/e04c0885-830a-479b-844b-4c7af79697f8/en-US/lab-0/2-on-your-own/5-update-workspace) to disable the temporary credentials settings.

**Comment by mervinyan** [comment]
Hi @ujwalbukka 

Good news! The error is gone and the stack was deployed successfully!

First I followed the instructions listed in above links but it didn't work due to "invalid endpoint: https://s3.amazonaws.com" error as I removed the credentials file from .aws folder per the below instruction 
![image](https://github.com/aws-samples/aws-serverless-saas-workshop/assets/1260845/e7376469-2ca2-4d88-a75c-bbeb05d57f7b)

I re-ran the "aws configure" command to reset the access key and security access key then re-ran the deployment scripts all steps completed successfully.

Many thanks for your help!

Have a nice weekend!
M
